pkg/provision/gce.go

```diff
@@ -158,18 +161,77 @@ func (gce *GCEProvisioner) createInletsFirewallRule(projectID string, firewallRu
 }
 
 // Delete deletes the GCE exit node
-func (gce *GCEProvisioner) Delete(id string) error {
-	instanceName, zone, projectID, err := getGCEFieldsFromID(id)
-	if err != nil {
-		return fmt.Errorf("Could not get custom GCE fields: %v", err)
+func (gce *GCEProvisioner) Delete(config HostDeleteConfig) error {
+	var instanceName string
+	var err error
+	if len(config.IP) > 0 {
+		inletID, err := gce.lookupID(config)
+		if err != nil {
+			return err
+		}
+		instanceName, _, _, err = getGCEFieldsFromID(inletID)
+		if err != nil {
+			return err
+		}
 	}
-	_, err = gce.gceProvisioner.Instances.Delete(projectID, zone, instanceName).Do()
+	_, err = gce.gceProvisioner.Instances.Delete(config.ProjectID, config.Zone, instanceName).Do()
 	if err != nil {
 		return fmt.Errorf("Could not delete the GCE instance: %v", err)
 	}
 	return err
 }
 
+// List returns a list of exit nodes
+func (gce *GCEProvisioner) List(filter ListFilter) ([]*ProvisionedHost, error) {
+	var inlets []*ProvisionedHost
+	var pageToken string
+	for {
+		call := gce.gceProvisioner.Instances.List(filter.ProjectID, filter.Zone).Filter(filter.Filter)
+		if len(pageToken) > 0 {
+			call = call.PageToken(pageToken)
+		}
+
+		instances, err := call.Do()
+		if err != nil {
+			return inlets, fmt.Errorf("Could not list instances: %v", err)
+		}
+		for _, instance := range instances.Items {
+			var status string
+			if instance.Status == "RUNNING" {
```

Comment:
We must use the `RUNNING` text in more than once place for GCP, can we extract a named const?